specification/agricultureplatform/AgriculturePlatform.DataPlane/models.tsp

```diff
@@ -0,0 +1,2140 @@
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  createdDateTime?: utcDateTime;
+
+  /**
+   * Date-time when resource was last modified, sample format: yyyy-MM-ddTHH:mm:ssZ.
+   */
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  modifiedDateTime?: utcDateTime;
+
+  /**
+   * Source of the resource.
+   */
+  @maxLength(100)
+  @minLength(2)
+  source?: string;
+
+  /**
+   * Name to identify resource.
+   */
+  @maxLength(100)
+  name?: string;
+
+  /**
+   * Textual description of the resource.
+   */
+  @maxLength(500)
+  description?: string;
+
+  /**
+   * Created by user/tenant id.
+   */
+  @visibility("read")
+  createdBy?: string;
+
+  /**
+   * Modified by user/tenant id.
+   */
+  @visibility("read")
+  modifiedBy?: string;
+
+  /**
+   * A collection of key value pairs that belongs to the resource.
+   * Each pair must not have a key greater than 50 characters
+   * and must not have a value greater than 150 characters.
+   * Note: A maximum of 25 key value pairs can be provided for a resource and only
+   * string,
+   * numeral and datetime (yyyy-MM-ddTHH:mm:ssZ) values are supported.
+   */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Should use Record<unknown> to model Map<String,Object>"
+  properties?: Record<unknown>;
+}
+
+/**
+ * ItemResponse model for Items api.
+ */
+model ItemResponse {
+  /**
+   * Links.
+   */
+  links?: Link[];
+
+  /**
+   * Schema for records under the dataset.
+   */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Should use Record<unknown> to model Map<String,Object>"
+  data: Record<unknown>;
+
+  /**
+   * OwnerItemId.
+   */
+  ownerItemId?: string;
+
+  /**
+   * CatalogId.
+   */
+  @visibility("read")
+  catalogId?: string;
+
+  /**
+   * CollectionId.
+   */
+  @visibility("read")
+  collectionId?: string;
+
+  /**
+   * Assets.
+   */
+  assets?: Asset[];
+
+  /**
+   * Unique resource ID.
+   */
+  @visibility("read")
+  id?: string;
+
+  /**
+   * The ETag value to implement optimistic concurrency.
+   */
+  @visibility("read")
+  eTag?: string;
+
+  /**
+   * Status of the resource.
+   */
+  @maxLength(100)
+  status?: string;
+
+  /**
+   * Date-time when resource was created, sample format: yyyy-MM-ddTHH:mm:ssZ.
+   */
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  createdDateTime?: utcDateTime;
+
+  /**
+   * Date-time when resource was last modified, sample format: yyyy-MM-ddTHH:mm:ssZ.
+   */
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  modifiedDateTime?: utcDateTime;
+
+  /**
+   * Source of the resource.
+   */
+  @maxLength(100)
+  @minLength(2)
+  source?: string;
+
+  /**
+   * Name to identify resource.
+   */
+  @maxLength(100)
+  name?: string;
```

Comment:
There are a couple of properties repeated across almost every model. Consider creating a common model with them and then spread them inside the resource models (...CommonProps)